### DISCUSSION TITLE
cancel() doesn't reset offset?

### DISCUSSION THREAD

**Discussion by cvlmtg** [pmndrs/use-gesture#673] [Q&A]
I'm trying to implement some drag and drop code with long press on mobile devices. I'm not sure if this is a bug or something I'm doing wrong, but I'll try to explain as best as I can:

I'm using `@react-spring/web` version 9.7.3 and `@use-gesture/react` version 10.3.1. my code should be fairly standard stuff taken straight from the examples, the only difference is a timer which starts `onTouchStart`. if the user moves a certain offset before the timer ends, I call `cancel()`. after calling cancel, if I move after the long press, my div "jumps" to the last offset. i.e.:

- I move after the timer from [ 0, 0 ] to offset [ 60, 20 ]
  - offset is [ 60, 20 ], lastOffset is [ 0, 0 ]
- I move before the timer to [ 70, 100 ] and my gesture is `cancel`ed()
  - offset is [ 70, 100 ], lastOffset is [ 60, 20 ]
- I move after the timer
  - offset is [ 70, 100 ], lastOffset is [ 70, 100 ]

so basically what I see is my div jumping from [ 60, 20 ]. to [ 70, 100 ]. since I canceled that gesture, I expect to have a new gesture having offset equal to [ 60, 20 ], where my div is currently positioned.
The reason I use `offset` is because react-spring is moving the div with `transform: translate3d();`

Any way to solve this?
Thanks!

**Comment by shawtung** [comment]
```typescript
useDrag(
  ({ offset: [x, y] }) => {
      api.start({ x, y });
  },
  {
    target: ref,
    from: () => [style.x.get(), style.y.get()],
  },
);
```
